### PR TITLE
fix: enable errcheck and handle all errors

### PR DESCRIPTION
**Pull request by jordivilaseca**
## Describe your changes
- Log when there is an error configuring the event generation logic

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have made the appropriate changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] If it is a major user facing functionality, I have added behavioural tests
- [x] If it is a critical part of the code or of significant complexity, I have added thorough testing

### CODE REVIEW

File: dataplane/processor.go [hunk 1]
```diff
@@ -229,7 +229,10 @@ func (p *Processor) UpdateConfig(resource, sampler string, config *control.Sampl
 			}
 		}
 		if eventor != nil {
-			eventor.SetEventsConfig(config.Events)
+			err := eventor.SetEventsConfig(config.Events)
```
Comment (GerardGarcia) [inline dataplane/processor.go:232]: we should configure a static check in the github pipeline that alerts us of unhandled errors

File: dataplane/processor.go [hunk 2]
```diff
@@ -229,7 +229,10 @@ func (p *Processor) UpdateConfig(resource, sampler string, config *control.Sampl
...
+			if err != nil {
+				logger.Error("Could not set event config", "error", err)
+			}
 			tr.eventor = eventor
```
Comment (GerardGarcia) [inline dataplane/processor.go:236]: should be still assign it if there was an error setting its config?

Comment (jordivilaseca) [reply dataplane/processor.go:236]: Good point, that could be problematic

File: dataplane/processor.go (hunk 1 above)
Comment (jordivilaseca) [reply dataplane/processor.go:232]: Done!

**Review by GerardGarcia — APPROVED** [review]

**Pull request merged**